gobblin-runtime/src/main/java/gobblin/runtime/DbDatasetStateStore.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.runtime;
+
+import com.google.common.base.CharMatcher;
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+import com.mysql.jdbc.jdbc2.optional.MysqlDataSource;
+import gobblin.annotation.Alias;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.metastore.DatasetStateStore;
+import gobblin.metastore.DbStateStore;
+import gobblin.password.PasswordManager;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import javax.sql.DataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A custom extension to {@link DbStateStore} for storing and reading {@link JobState.DatasetState}s.
+ *
+ * <p>
+ *   The purpose of having this class is to hide some implementation details that are unnecessarily
+ *   exposed if using the {@link DbStateStore} to store and serve job/dataset states between job runs.
+ * </p>
+ *
+ * <p>
+ *   In addition to persisting and reading {@link JobState.DatasetState}s. This class is also able to
+ *   read job state files of existing jobs that store serialized instances of {@link JobState} for
+ *   backward compatibility.
+ * </p>
+ *
+ */
+public class DbDatasetStateStore extends DbStateStore<JobState.DatasetState>
+    implements DatasetStateStore<JobState.DatasetState> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DbDatasetStateStore.class);
+
+  @Alias("MySqlDbDatasetStateStore")
```

Comment:
nit: I'd change it to just "mysql" . The alias name scope is per interface so there should not be conflicts.